tictactoe/adapter-vdm/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/adapter/vdm/check/PlayerCanPlaceTokenCheck.kt

```diff
@@ -7,31 +7,26 @@ import com.anaplan.engineering.azuki.tictactoe.adapter.vdm.toVdmPlayer
 import com.anaplan.engineering.azuki.tictactoe.adapter.vdm.toVdmPos
 import com.anaplan.engineering.azuki.vdm.DefaultModuleBuilder
 
-class PlayerCannotPlaceTokenCheck(
-    private val gameName: String,
-    private val playerName: String,
-    private val position: Position
+class PlayerCanPlaceTokenCheck(
+    private val gameName: String, private val playerName: String, private val position: Position
 ) : PlaceATokenBehaviour(), DefaultVdmCheck {
 
     override fun build(builder: DefaultModuleBuilder): DefaultModuleBuilder {
         val gameGetter = builder.getters[gameName] ?: throw IllegalStateException("Missing getter for game $gameName")
 
-        return builder.extend(
-            requiredImports = setOf(
-                XOModule.Game.import,
-                XOModule.Pos.import,
-                XOModule.move.import,
-            ),
-            failingStep = """
+        return builder.extend(requiredImports = setOf(
+            XOModule.Game.import,
+            XOModule.Player.import,
+            XOModule.Pos.import,
+            XOModule.isValidMove.import,
+        ), testSteps = listOf("""
                 (
                     dcl g: ${XOModule.Game} := $gameGetter;
+                    dcl p: ${XOModule.Player} := ${toVdmPlayer(playerName)};
                     dcl pos: ${XOModule.Pos} := ${toVdmPos(position)};
-                    let
-                        g' = ${XOModule.move}(g, ${toVdmPlayer(playerName)}, pos)
-                    in
-                        return false
+                    dcl expected: bool := true;
+                    ${checkEquals(actual = "${XOModule.isValidMove}(g, p, pos)")}
                 );
-            """
-            )
+            """))
```

Comment:
As above, this is _not_ a rewrite of the LHS, this is a new check entirely.

Is this check correct?  I'm not very good at VDM.

Reply:
I did, yeah.  (I think I did the same in Kazuki.)  I can back these out if needed but I presume that they're not problems per se?